### PR TITLE
st1 [2527][ADD] purchase_channel_attribute

### PR DESCRIPTION
**Pull request by kanda999**
[2527](https://www.quartile.co/web?#id=2527&action=771&model=project.task&view_type=form&menu_id=505)

add request_medium_id purchase_category_id shop_id in purchase.order

### CODE REVIEW

File: purchase_add_info/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2022 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+{
+    "name": "Purchase Add Info",
+    "version": "15.0.1.0.0",
+    "author": "Quartile Limited",
+    "website": "https://www.quartile.co",
+    "category": "Purchase",
+    "license": "LGPL-3",
+    "depends": ["hr", "purchase"],
```
Comment (yostashiro) [inline purchase_add_info/__manifest__.py]: Why does this depend on hr?

File: purchase_channel_attribute/models/purchase_category.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    name = fields.Char(required=True)
+    sequence = fields.Integer(default=10)
+    active = fields.Boolean(default=True)
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        default=lambda self: self.env["res.company"]._company_default_get(
+            "request.medium"
+        ),
```
Comment (yostashiro) [inline purchase_channel_attribute/models/purchase_category.py]:
```suggestion
        default=lambda self: self.env.company,
```
`_company_default_get()` has been deprecated.

File: purchase_channel_attribute/models/request_medium.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    name = fields.Char(required=True)
+    sequence = fields.Integer(default=10)
+    active = fields.Boolean(default=True)
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        default=lambda self: self.env["res.company"]._company_default_get(
+            "request.medium"
+        ),
```
Comment (yostashiro) [inline purchase_channel_attribute/models/request_medium.py]:
```suggestion
        default=lambda self: self.env.company,
```

File: purchase_channel_attribute/data/menuitem_data.xml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <menuitem
+        id="menu_purchase_in_config_purchase"
+        name="Purchase"
```
Comment (yostashiro) [inline purchase_channel_attribute/data/menuitem_data.xml]:
```suggestion
        name="Purchases"
```

**Review by yostashiro — APPROVED** [review]
@kanda999 Please fix the commented parts and go ahead.

**Comment by kanda999** [comment]
add request_medium_id purchase_category_id shop_id in purchase.order

**Pull request merged**